Mathlib/RingTheory/Trace/Basic.lean

```diff
@@ -526,46 +529,108 @@ theorem Algebra.trace_surjective [FiniteDimensional K L] [Algebra.IsSeparable K
   rw [LinearMap.range_eq_bot]
   exact Algebra.trace_ne_zero K L
 
-variable {K L}
+end DetNeZero
+
+section isNilpotent
+
+namespace Algebra
+
+/-- The trace of a nilpotent element is nilpotent. -/
+lemma trace_isNilpotent_of_isNilpotent {R S : Type*} [CommRing R] [CommRing S] [Algebra R S] {x : S}
```

Comment:
I would call this `Algebra.isNilpotent_trace_of_isNilpotent` to be compatible with the `LinearMap` version.

Reply:
Ah sorry, you are just moving the declaration... still the two names should be consistent, and technically that statement is `IsNilpotent (trace ...)`, so the `LinearMap` version looks correct (can you also add a deprecation?).